File: image_processing/01_images/Python/02_imagePGM_modular/02_image_main.py
from ImagePGM import *

# Main function
if __name__ == "__main__":
    # Read the input PGM image
    image = ImagePGM("../../../_data/robot.pgm")

    print("Infos")
    print(image)
    
    image.writeImagePGM("robot2.pgm")
    print("End of test")

File: image_processing/01_images/02_imagePGM_modular/02_image_main.cpp
#include <iostream>
#include <string>
#include <vector>
#include <fstream>
#include "ImagePGM.h"


int main(int argc, char** argv) {
    // Read the input PGM image
    ImagePGM image("../../_data/robot.pgm");

	std::cout << "Infos" << std::endl;
	std::cout << image << std::endl;

	if(image.writeImagePGM("robot2.pgm")){
		std::cout << "Image written" << std::endl;
	}

    return 0;
}
